Source/Scene/Model.js

```diff
@@ -1454,6 +1474,371 @@ define([
         return getStringFromTypedArray(loadResources.getBuffer(bufferView));
     }
 
+    function replaceAllButFirstInString(string, find, replace) {
+        return string.replace(new RegExp(find, 'g'), function(match, offset, all) {
+            return all.indexOf(find) === offset ? match : replace;
+        });
+    }
+
+    function scaleFromMatrix5Array(matrix) {
+        return [matrix[0], matrix[1], matrix[2], matrix[3],
+                matrix[5], matrix[6], matrix[7], matrix[8],
+                matrix[10], matrix[11], matrix[12], matrix[13],
+                matrix[15], matrix[16], matrix[17], matrix[18]];
+    }
+
+    function translateFromMatrix5Array(matrix) {
+        return [matrix[20], matrix[21], matrix[22], matrix[23]];
+    }
+
+    function stringifyArray(array) {
+        return array.map(function(val) {
+            return val.toFixed(20);
+        });
+    }
+
+    // Makes a copy of the given material, its technique, and its program,
+    // and adds them to the gltf hierarchy with new ids
+    // returns the id of the new material
+    function splitMaterial(model, materialId) {
+        var gltf = model.gltf;
+        var num = 0;
+
+        var material = gltf.materials[materialId];
+        var newMaterialId;
+
+        // Find an available name for the new material
+        while (!defined(newMaterialId) ||
```

Comment:
A `do...while` may also be a bit cleaner and avoid the `!defined(newMaterialId)` check.